homeassistant/components/alarm_control_panel/__init__.py

```diff
@@ -52,13 +59,24 @@ def setup(hass, config):
     def alarm_service_handler(service):
         """Map services to methods on Alarm."""
         target_alarms = component.extract_from_service(service)
-
         code = service.data.get(ATTR_CODE)
 
-        method = SERVICE_TO_METHOD[service.service]
```

Comment:
I think you need to keep this, as well as the getattr below. Then I'd look at a simple if statement like so:
If service.service == SERVICE_ALARM_KEYPRESS:
    ...get the keys to press from service data
    ...pass the result of that to getattr instead of "code"
....same for output control
....else:
    ...get the code value
    ...call the getattr line just like it is now

Make sense?  Sorry- on iPhone typing this.